widgetssdk/src/main/java/com/glia/widgets/survey/SurveyActivity.java

```diff
@@ -17,18 +18,21 @@ public class SurveyActivity extends AppCompatActivity {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-
         setContentView(R.layout.survey_activity);
+        prepareSurveyView();
+    }
 
+    private void prepareSurveyView() {
         surveyView = findViewById(R.id.survey_view);
 
         SurveyContract.Controller surveyController =
                 Dependencies.getControllerFactory().getSurveyController();
         surveyView.setController(surveyController);
-
         Bundle extras = getIntent().getExtras();
-        Survey survey = extras.getParcelable(GliaWidgets.SURVEY);
+        UiTheme uiTheme = extras.getParcelable(GliaWidgets.UI_THEME);
+        surveyView.setTheme(uiTheme);
 
+        Survey survey = extras.getParcelable(GliaWidgets.SURVEY);
```

Comment:
Would be better to get pending survey from repository instead
What do You think @jakob-mc

Reply:
A survey already gotten from the repository and that is reason to starting the SurveyActivity. Also, an engagement ends before the SurveyActivity started.

Reply:
Ill move this discussion under logic task:
https://github.com/salemove/android-sdk-widgets/pull/248/files#r859557863